### PR TITLE
process-agent,network-tracer: Move the main packages and the tooling

### PR DESCRIPTION
**Pull request by sfluor**
### What does this PR do?

This moves the main packages for the process-agent the network-tracer and the nettop util into this repository (`nettop` now lives in `pkg/ebpf`).

This PR also changes moves the old rake tasks defined here https://github.com/DataDog/datadog-process-agent/blob/master/Rakefile into invoke tasks.

There will be one more PR coming to add the process-agent and the network-tracer to the CI builds (the makefile for the eBPF code will also be removed and added to the invoke tasks in the next pr)

### Motivation

Merging the process-agent into this repository.

### Additional Notes

Advised to review commit by commit (the first one is pretty big since it's mostly imported code)

### CODE REVIEW

**Comment by leeavital** [comment]
Sorry if I missed this, but can we re-enable the test for that checks that the .o matches the go bindata ebpf module in this PR?

**Comment by sfluor** [comment]
(Per our discussion with @leeavital, I will add the test to check for the object files correctness in the next PR)

File: Dockerfiles/network-tracer/Dockerfile-ebpf [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Used to compile eBPF code
```
Comment (sunhay) [inline Dockerfiles/network-tracer/Dockerfile-ebpf]: Is there a better place we can move these dockerfiles? I'm not sure `/Dockerfiles` is meant for build pipeline specific dockerfiles (may be wrong)

File: tasks/network_tracer.py [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+        "REPO_PATH": REPO_PATH,
+    }
+
+    ctx.run(cmd.format(**args), env=env)
+
+
+@task
+def test(ctx):
+    """
+    Run tests on eBPF parts
```
Comment (sunhay) [inline tasks/network_tracer.py]: Do these tests run in an environment where we can run the actual eBPF code?

File: tasks/network_tracer.py [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+
+    files = glob.glob("pkg/ebpf/c/*.[c,h]")
+
+    for file in files:
+        ctx.run(fmtCmd.format(file=file))
+        ctx.run(sedCmd.format(file=file))
+
+
+@task
+def build_docker_image(ctx, image_name):
```
Comment (sunhay) [inline tasks/network_tracer.py]: `build_dev_docker_image`?

File: tasks/network_tracer.py [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+
+
+def build_object_files(ctx):
+    """build_object_files_only builds only the eBPF object
+    (without rebuilding the docker image builder)
+    """
+
+    makeCmd = "make -f /ebpf/c/tracer-ebpf.mk build install"
+    args = {
+        "circle_url": "TODO",
```
Comment (sunhay) [inline tasks/network_tracer.py]: Is this coming in another PR?

File: tasks/process_agent.py [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+
+
+@task
+def build(ctx, race=False, incremental_build=False, use_embedded_libs=False, puppy=False, use_venv=False):
+    """
+    Build the process agent
+    """
+
+    # generate windows resources
+    if sys.platform == 'win32':
```
Comment (sunhay) [inline tasks/process_agent.py]: Have you done any testing around the windows binary built by this yet?

File: cmd/process-agent/README.md [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+
+## Installation
+
+See the [Live Processes docs](https://docs.datadoghq.com/graphing/infrastructure/process/#installation) for installation instructions.
+
+## Development or running from source
+
+Pre-requisites:
+
+* `go >= 1.10.1`
```
Comment (sunhay) [inline cmd/process-agent/README.md]: Will this whole change set make us use a different version of go now?

File: pkg/process/tools/dev_setup.sh [hunk 7]
```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env bash
```
Comment (sunhay) [inline pkg/process/tools/dev_setup.sh]: `pkg/process/tools` seems like an odd place to put this. Is there anywhere else outside of `pkg` we can move it instead?

File: Dockerfiles/network-tracer/Dockerfile-ebpf (hunk 1 above)
Comment (sfluor) [reply Dockerfiles/network-tracer/Dockerfile-ebpf]: 👍  I can move them into the `tools` directory under `ebpf`

File: tasks/network_tracer.py (hunk 2 above)
Comment (sfluor) [reply tasks/network_tracer.py]: Not yet but this is planned on the next PR

File: tasks/network_tracer.py [hunk 8]
```diff
@@ -44,7 +44,7 @@ def build(ctx, race=False, rebuild_ebpf_builder=False, incremental_build=False,
 
     # Add custom ld flags
     ldflags += ' '.join(["-X '{name}={value}'".format(name=main+key, value=value) for key, value in ld_vars.items()])
-    build_tags = get_default_build_tags(puppy=puppy).append(BPF_TAG)
+    build_tags = get_default_build_tags(puppy=puppy) + [BPF_TAG]
 
     # TODO static option
```
Comment (sunhay) [inline tasks/network_tracer.py]: Coming in another PR? Possibly being left out if we don't need static builds?

File: tasks/process_agent.py (hunk 5 above)
Comment (sfluor) [reply tasks/process_agent.py]: Not yet I don't really have a good windows environment for this, but I will test the binaries built from the regular pipeline in the next PR 👍

File: cmd/process-agent/README.md (hunk 6 above)
Comment (sfluor) [reply cmd/process-agent/README.md]: Good catch I will update this 👍

File: tasks/network_tracer.py (hunk 8 above)
Comment (sfluor) [reply tasks/network_tracer.py]: Mostly left out if we don't need static builds, if we do we could just import this part too: https://github.com/DataDog/datadog-process-agent/blob/master/gorake.rb#L58-L72

File: pkg/process/tools/dev_setup.sh (hunk 7 above)
Comment (sfluor) [reply pkg/process/tools/dev_setup.sh]: 👍  will also move it under the tools dir

File: tasks/network_tracer.py (hunk 4 above)
Comment (sfluor) [reply tasks/network_tracer.py]: I will remove it 👍

**Review by sunhay — DISMISSED** [review]
Looks great on my side. Definitely going to need eyes from the core-agent team regarding the build pieces :)

File: tasks/utils.py [hunk 9]
```diff
@@ -136,6 +136,18 @@ def get_version_ldflags(ctx, prefix=None):
     ldflags += "-X {}/pkg/serializer.AgentPayloadVersion={} ".format(REPO_PATH, payload_v)
     return ldflags
 
+def get_git_commit():
```
Comment (albertvaka) [inline tasks/utils.py]: Maybe use this new function from `get_version_ldflags` above, so there is a bit less of duplication.

**Review by albertvaka — DISMISSED** [review]
I have only reviewed the invoke tasks, but it looks good to me.

File: tasks/utils.py (hunk 9 above)
Comment (albertvaka) [reply tasks/utils.py]: Feel free to ignore this or change it in the next PR so we don't block the merge even longer :)

Comment (sfluor) [reply tasks/utils.py]: 👍

**Review by albertvaka — APPROVED** [review]

**Review by sunhay — APPROVED** [review]

**Pull request merged**